### PR TITLE
Feat/seven discard

### LINKED ISSUE #1112: [Feature]: Implement Seven Double Jack special case in the GameState API
### Feature Summary

We need to implement discarding a jack from a seven in the special case where both the top cards are jacks which can’t be played 

### Detailed Description

#1111 implemented MoveType.SEVEN_JACK in the usual case, but excluded the special situation where the top two cards are jacks which can’t be played. Because both the validate() and execute() logic are significantly different from the “normal” sevenJack case, we should implement this as its own (new) moveType: MoveType.SEVEN_DISCARD. 

In addition to creating the usual validate() and execute() helpers, this will require updating MoveType, adding a new case to hasValidMoveBody and updating the API code in the game store and cypress command file to dynamically request the correct endpoint + MoveType based on the env var

### PR DESCRIPTION
**Pull request by itsalaidbacklife**
Implements seven discard: throwing away a jack from the top of the deck when both cards are unplayable jacks

## Issue number

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #1112 

## Please check the following

- [ ] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [ ] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

### CODE REVIEW

File: tests/e2e/support/commands.js [hunk 1]
```diff
@@ -881,13 +879,8 @@ Cypress.Commands.add('playJackFromSevenOpponent', (card, target) => {
...
 
       const cardId = foundCard.id;
-      let targetId;
+      const targetId = foundTarget.id;
 
-      if (target !== -1) {
-        targetId = foundTarget.id;
-      } else {
-        targetId = -1;
-      }
```
Comment (itsalaidbacklife) [inline tests/e2e/support/commands.js:890]: Moved `cy. sevenDiscardOpponent()` into it's own command and simplified this command (`cy. playJackFromSevenOpponent()` so it's only for actually playing the jack

File: api/helpers/game-states/moves/seven-discard/execute.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Discard a Jack via a seven',
+
+  description: 'Returns new GameState resulting discarding the played card (due to being unable to play card to resolve a previous seven, when both of the top two cards were jacks which could not be played)',
```
Comment (Haviles04) [inline api/helpers/game-states/moves/seven-discard/execute.js:6]:
```suggestion
  description: 'Returns new GameState resulting in discarding the played card (due to being unable to play card to resolve a previous seven, when both of the top two cards were jacks which could not be played)',
```

**Comment by Haviles04** [comment]
Nice, I think we need to update `getLog` as well since we added a movetype

**Review by Haviles04 — APPROVED** [review]
LGTM :)

**Pull request merged**